cartography/intel/azure/key_vaults.py

```diff
@@ -0,0 +1,293 @@
+                    "updated_on": key_props.updated_on,
+                }
+            )
+        return keys
+    except (ClientAuthenticationError, HttpResponseError) as e:
+        logger.warning(f"Failed to get keys for vault {vault_uri}: {str(e)}")
+        return []
+
+
+@timeit
+def get_certificates(credentials: Credentials, vault_uri: str) -> list[dict]:
+    try:
+        client = CertificateClient(
+            vault_url=vault_uri, credential=credentials.credential
+        )
+        certs = []
+        for cert_props in client.list_properties_of_certificates():
+            certs.append(
+                {
+                    "id": cert_props.id,
+                    "name": cert_props.name,
+                    "enabled": cert_props.enabled,
+                    "created_on": cert_props.created_on,
+                    "updated_on": cert_props.updated_on,
+                    "x5t": (
+                        cert_props.x509_thumbprint.hex()
+                        if cert_props.x509_thumbprint
+                        else None
+                    ),
+                }
+            )
+        return certs
+    except (ClientAuthenticationError, HttpResponseError) as e:
+        logger.warning(f"Failed to get certificates for vault {vault_uri}: {str(e)}")
+        return []
+
+
+@timeit
+def transform_key_vaults(key_vaults_response: list[dict]) -> list[dict]:
+    transformed_vaults: list[dict[str, Any]] = []
+    for vault in key_vaults_response:
+        transformed_vault = {
+            "id": vault.get("id"),
+            "name": vault.get("name"),
+            "location": vault.get("location"),
+            "tenant_id": vault.get("properties", {}).get("tenant_id"),
+            "sku_name": vault.get("properties", {}).get("sku", {}).get("name"),
+            "vault_uri": vault.get("properties", {}).get("vault_uri"),
+        }
+        transformed_vaults.append(transformed_vault)
+    return transformed_vaults
+
+
+@timeit
+def transform_secrets(secrets_response: list[dict]) -> list[dict]:
+    transformed_secrets: list[dict[str, Any]] = []
+    for secret in secrets_response:
+        transformed_secret = {
+            "id": secret.get("id"),
+            "name": secret.get("name"),
+            "enabled": secret.get("enabled"),
+            "created_on": secret.get("created_on"),
+            "updated_on": secret.get("updated_on"),
+        }
+        transformed_secrets.append(transformed_secret)
+    return transformed_secrets
+
+
+@timeit
+def transform_keys(keys_response: list[dict]) -> list[dict]:
+    transformed_keys: list[dict[str, Any]] = []
+    for key in keys_response:
+        transformed_key = {
+            "id": key.get("id"),
+            "name": key.get("name"),
+            "enabled": key.get("enabled"),
+            "created_on": key.get("created_on"),
+            "updated_on": key.get("updated_on"),
+        }
+        transformed_keys.append(transformed_key)
+    return transformed_keys
+
+
+@timeit
```

Comment:
No need the timeit for a transform since there's no IO over the network or database